packages/zendesk-adapter/src/adapter.ts

```diff
@@ -185,6 +222,7 @@ export default class ZendeskAdapter implements AdapterOperations {
     this.configInstance = configInstance
     this.getElemIdFunc = getElemIdFunc
     this.client = client
+    this.credentials = credentials
```

Comment:
after looking at the function below, I think instead of saving the credentials on the adapter we should instead expose only the function for creating the client:
```suggestion
    this.createClientBySubdomain = (subdomain: string): ZendeskClient => {
      return new ZendeskClient({
        credentials: { ...this.credentials, subdomain },
        config: this.userConfig[CLIENT_CONFIG],
      })    
    }
```